packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/mocks.tsx

```diff
@@ -61,65 +63,57 @@ export const MOCK_RESET_DATA = {
   verified: true,
 };
 
-export function mockUrlQueryData(
-  params: Record<string, string> = mockCompleteResetPasswordParams
-) {
-  const window = new ReachRouterWindow();
-  const data = new UrlQueryData(window);
-  for (const param of Object.keys(params)) {
-    data.set(param, params[param]);
-  }
-  return data;
-}
+export const Subject = ({
+  integrationType = IntegrationType.Web,
+  params = mockCompleteResetPasswordParams,
+}: {
+  integrationType?: IntegrationType;
+  params?: Record<string, string>;
+}) => {
+  const urlQueryData = mockUrlQueryData(params);
 
-class StorageDataMock extends StorageData {
-  public override persist(): void {
-    // no op
-  }
-  public override load(): void {
-    // no op
+  let completeResetPasswordIntegration: CompleteResetPasswordIntegration;
+  switch (integrationType) {
+    case IntegrationType.OAuth:
+      completeResetPasswordIntegration =
+        createMockResetPasswordOAuthIntegration();
+      break;
+    case IntegrationType.SyncDesktop:
+      completeResetPasswordIntegration = createMockSyncDesktopIntegration();
+      break;
+    case IntegrationType.Web:
+    default:
+      completeResetPasswordIntegration = createMockWebIntegration();
   }
-  public override set(): void {
-    // no op
-  }
-}
 
-const route = '/complete_reset_password';
-export const getSubject = (
-  account: Account,
-  params?: Record<string, string>
-) => {
-  const urlQueryData = mockUrlQueryData(params);
-  const history = createHistoryWithQuery(
-    route,
-    new URLSearchParams(params).toString()
+  return (
+    <LinkValidator
+      linkType={LinkType['reset-password']}
+      viewName={'complete-reset-password'}
+      getParamsFromModel={() => {
+        return new CompleteResetPasswordLink(urlQueryData);
+      }}
+      // TODO worth fixing this type?
```

Comment:
I think it is. This integration type is actually much broader than what link validator needs. LinkValidator really just needs `LinkExpiredResetPasswordIntegration` to be fullfilled.

Reply:
I have this TODO in two places. I'll punt doing this out from this PR since you're going to take a look at `LinkValidator` (though I think it'd be easy enough to fix).